src/domains/heap_domain.cpp

```diff
@@ -98,7 +99,10 @@ exprt heap_domaint::value_to_ptr_exprt(const exprt &expr)
   {
     const std::string value=id2string(to_constant_expr(expr).get_value());
     if(value.substr(value.size()/2).find_first_not_of('0')!=std::string::npos)
-      return plus_exprt(expr.op0(), constant_exprt::integer_constant(0));
+    {
+      constant_exprt zero=constant_exprt("0", integer_typet());
```

Comment:
There's a function `make_zero` in `domains/util.h` that could be used here. It cannot handle `integer_typet` now, but it should be easy to extend it.

Then this could be: `make_zero(integer_typet())`.

In fact, I'm not even sure why there's `integer_typet` here as we use bitvectors everywhere. I guess that `unsignedbv_typet` should work as well.